core/src/qrcode/QRDecoder.cpp

```diff
@@ -316,17 +321,28 @@ DecoderResult DecodeBitStream(ByteArray&& bytes, const Version& version, ErrorCo
 
 DecoderResult Decode(const BitMatrix& bits)
 {
-	const Version* pversion = ReadVersion(bits);
+	bool isModel1 = false;
+
+	const Version* pversion = ReadVersion(bits, isModel1);
 	if (!pversion)
 		return FormatError("Invalid version");
-	const Version& version = *pversion;
+	const Version& preversion = *pversion;
 
-	auto formatInfo = ReadFormatInformation(bits, version.isMicroQRCode());
+	auto formatInfo = ReadFormatInformation(bits, preversion.isMicroQRCode(), isModel1);
 	if (!formatInfo.isValid())
 		return FormatError("Invalid format information");
 
+	if (isModel1)
```

Comment:
This split between reading version and reading format info is obviously not suitable anymore when those two inter-depend in the presence of Model1. I'll think about how I'd like this to be cleanup after merging.